pySDC/projects/Second_orderSDC/stability_simulation.py

```diff
@@ -0,0 +1,287 @@
+
+        return stab_RKN
+
+    def plot_stability(self, region, title=""):  # pragma: no cover
+        """
+        Plotting runtine for moduli
+
+        Args:
+            stabval (numpy.ndarray): moduli
+            title: title for the plot
+        """
+        set_fixed_plot_params()
+        lam_k_max = np.amax(self.lambda_kappa)
+        lam_mu_max = np.amax(self.lambda_mu)
+
+        plt.figure()
+        levels = np.array([0.25, 0.5, 0.75, 0.9, 1.0, 1.1])
+
+        CS1 = plt.contour(self.lambda_kappa, self.lambda_mu, region.T, levels, colors='k', linestyles="dashed")
+        # CS2 = plt.contour(self.lambda_k, self.lambda_mu, np.absolute(region.T), [1.0], colors='r')
+
+        plt.clabel(CS1, inline=True, fmt="%3.2f")
+
+        plt.gca().set_xticks(np.arange(0, int(lam_k_max) + 3, 3))
+        plt.gca().set_yticks(np.arange(0, int(lam_mu_max) + 3, 3))
+        plt.gca().tick_params(axis="both", which="both")
+        plt.xlim([0.0, lam_k_max])
+        plt.ylim([0.0, lam_mu_max])
+
+        plt.xlabel(r"$\Delta t\cdot \kappa$", labelpad=0.0)
+        plt.ylabel(r"$\Delta t\cdot \mu$", labelpad=0.0)
+        if self.RKN:
+            plt.title(f"{title}")
+        if self.radius:
+            plt.title("{}  $M={}$".format(title, self.num_nodes))
+        else:
+            plt.title(r"{}  $M={},\  K={}$".format(title, self.num_nodes, self.K_iter))
+        plt.tight_layout()
+        plt.savefig(self.cwd + "data/M={}_K={}_redion_{}.pdf".format(self.num_nodes, self.K_iter, title))
+
+    def run_SDC_stability(self):  # pragma: no cover
+        self.RKN = False
+        self.radius = False
+        self.plot_stability(self.SDC, title="SDC stability region")
+
+    def run_Picard_stability(self):  # pragma: no cover
+        self.RKN = False
+        self.radius = False
+        self.plot_stability(self.picard, title="Picard stability region")
+
+    def run_Ksdc(self):  # pragma: no cover
+        self.radius = True
+        self.plot_stability(self.Ksdc, title="$K_{sdc}$ spectral radius")
+
+    def run_Kpicard(self):  # pragma: no cover
+        self.radius = True
+        self.plot_stability(self.Kpicard, title="$K_{picard}$ spectral radius")
+
+    def run_RKN_stability(self):  # pragma: no cover
+        self.RKN = True
+        self.radius = False
+        region_RKN = self.stability_data_RKN()
+        self.plot_stability(region_RKN.T, title='RKN-4 stability region')
+
+
+def compute_and_check_stability(description,
+    point, compute_interval=False, save_interval_file=False, interval_filename='./data/stab_interval.txt',
+    check_stability=False, store_results=False, results_filename='./data/stab_results.txt', quadrature_list=['GAUSS', 'LOBATTO']
+):
+
+    # Storage for stability interval and stability check
+    interval_data = []
+    results_data = []
+
+    # Loop through different numbers of nodes and maximum iterations
+    for num_nodes in description['helper_params']['num_nodes_list']:
```

Comment:
Why is `helper_params` not a separate parameter to this function? Since a `description` dictionary is heavily used in pySDC, I find this confusing. You expect everything in the description to be a parameter for pySDC, but here it's just used for this function. Hiding this necessary parameter for the function in the description object makes it difficult to understand what's going on.